modules/compute/src/main/java/org/apache/ignite/internal/compute/JobExecutionFutureDelegate.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.compute;
+
+import static org.apache.ignite.internal.lang.IgniteExceptionMapperUtil.convertToPublicFuture;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.compute.JobExecution;
+import org.apache.ignite.compute.JobStatus;
+
+/**
+ * Delegates {@link JobExecution} to the future of {@link JobExecution} wrapping exceptions to public.
+ *
+ * @param <R> Result type.
+ */
+class JobExecutionFutureDelegate<R> implements JobExecution<R> {
```

Comment:
I think the naming here is a little bit confusing. When I see `delegate` I think it is a delegate that *does the job* but in fact, this is a wrapper that *delegates the job*.  

I think the "delegate" itself is not the best postfix for the class but it is good to name the variable inside this class.

Reply:
Maybe a "wrapper" is a better choice.